src/org/rascalmpl/values/RascalFunctionValueFactory.java

```diff
@@ -583,7 +584,14 @@ private ITree parseObject(String methodName, ISourceLocation location, char[] in
                 recoverer = new ToTokenRecoverer(uri, parserInstance, new StackNodeIdDispenser(parserInstance));
                 //debugListener = new DebugLogger(new PrintWriter(System.out, true));
             }
-            return (ITree) parserInstance.parse(methodName, uri, input, exec, new DefaultNodeFlattener<>(), new UPTRNodeFactory(allowAmbiguity), recoverer, debugListener);
+            ITree parseForest = (ITree) parserInstance.parse(methodName, uri, input, exec, new DefaultNodeFlattener<>(), new UPTRNodeFactory(allowRecovery || allowAmbiguity), recoverer, debugListener);
+
+            if (!allowAmbiguity && allowRecovery && (filters == null || filters.isEmpty())) {
```

Comment:
looks like something that should be explained in the documentation of `parseTree`

Reply:
Definitely. Ik will add this to the documentation.